FlowCrypt/Controllers/Inbox/InboxViewController.swift

```diff
@@ -452,10 +452,22 @@ extension InboxViewController: MsgListViewController {
                 tableNode.reloadData()
             } else {
                 state = .fetched(.byNumber(total: newTotalNumber))
-                tableNode.deleteRows(at: [IndexPath(row: index, section: 0)], with: .left)
+                do {
+                    try ObjcException.catch {
+                        self.tableNode.deleteRows(at: [IndexPath(row: index, section: 0)], with: .left)
```

Comment:
Data source for table node should be updated with correct amount of `inboxInput` then reload should be performed.

Reply:
Data source is updated above:
```
inboxInput.remove(at: index)
```
We could reload table node with `reloadData()`, but crash means that we are in invalid state. We could move from index to some kind of message ID. 

But I guess that this PR is more about handling Objc exceptions in Swift.

Reply:
While originally it was indeed primary about handling of ObjC exceptions, that was in part because we didn't know any better.

If the actual root cause of this exception can be fixed, that is even better and more appropriate. Both should be done.

Reply:
I think it's something more specific than just updating data source, as this crash happens very rarely.
Implementing ObjC exception handling should help us to find root cause of this issue.

May be switching from index to message id (as @ivan-ushakov proposed) will fix this crash, but we can't be 100% sure as currently we're not able to reproduce this issue.

Reply:
If the goal is to collect more information about the crash, I suppose we should have it show a modal similar to other crash scenarios? Then some user may take a screenshot and send it to us for debugging.

As currently, it will just be logged then forgotten.